src/compas/geometry/xforms/quaternion.py

```diff
@@ -0,0 +1,225 @@
+
+
+from compas.geometry.transformations import quaternion_multiply
+from compas.geometry.transformations import quaternion_conjugate
+from compas.geometry.transformations import quaternion_unitize
+from compas.geometry.transformations import quaternion_canonize
+from compas.geometry.transformations import quaternion_norm
+from compas.geometry.transformations import quaternion_is_unit
+
+from compas.geometry.basic import allclose
+
+__all__ = ['Quaternion']
+
+
+class Quaternion(object):
+
+    """Creates a ``Quaternion`` object.
+
+    Parameters
+    ----------
+    w : float
+        The scalar (real) part of a quaternion.
+    x, y, z : float
+        Components of the vector (complex, imaginary) part of a quaternion.
+
+
+    Examples
+    --------
+    >>> Q = Quaternion(1.0, 1.0, 1.0, 1.0).unitized()
+    >>> R = Quaternion(0.0,-0.1, 0.2,-0.3).unitized()
+    >>> P = R*Q
```

Comment:
in this case there should be whitespace around the operator (it is not a case of operator precedence)